blocks/embed/embed.js

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+/* global window */
+/* eslint-disable import/named, import/extensions */
+
+import { buildFigure } from '../../scripts/scripts.js';
+
+const loadScript = (url, callback, type) => {
+    const $head = document.querySelector('head');
+    const $script = document.createElement('script');
+    $script.src = url;
+    if (type) {
+      $script.setAttribute('type', type);
+    }
+    $head.append($script);
+    $script.onload = callback;
+    return $script;
+}
+
+// 'open.spotify.com' returns 'spotify'
+const getServer = (url) => {
+    const l = url.hostname.lastIndexOf('.');
+    return url.hostname.substring(url.hostname.lastIndexOf('.', l - 1) + 1, l);
+}
+  
+const getDefaultEmbed = (url) => {
+    return `<div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;">
+      <iframe src="${url.href}" style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute;" allowfullscreen=""
+        scrolling="no" allow="encrypted-media" title="Content from ${url.hostname}" loading="lazy">
+      </iframe>
+    </div>`;
+}
+  
+const embedYoutube = (url) => {
+    const usp = new URLSearchParams(url.search);
+    const vid = usp.get('v');
+    const embed = url.pathname;
+    const embedHTML = `<div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;">
+      <iframe src="https://www.youtube.com${vid ? `/embed/${vid}?rel=0&amp;v=${vid}` : embed}" style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute;" allow="encrypted-media; accelerometer; gyroscope; picture-in-picture" allowfullscreen="" scrolling="no" title="Content from Youtube" loading="lazy"></iframe>
+    </div>`;
+    return embedHTML;
+}
+  
+const embedInstagram = (url) => {
+    const location = window.location.href;
+    const src = `${url.origin}${url.pathname}${url.pathname.charAt(url.pathname.length - 1) === '/' ? '' : '/'}embed/?cr=1&amp;v=13&amp;wp=1316&amp;rd=${location.endsWith('.html') ? location : `${location}.html`}`;
```

Comment:
i would pull out all the conditions `?` out and into separate lines as this is quite hard to read...

also there seems to be a mix of `url.pathname.charAt(url.pathname.length - 1)` and `location.endsWith('.html')` and i would probably only use `.endsWith()` as it seems more readable.

Reply:
I didn't actually write those code but I agree that there is readability issue. I will improve that.